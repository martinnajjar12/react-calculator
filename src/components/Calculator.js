import React, { useState } from 'react';
import { Grid, makeStyles } from '@material-ui/core';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import calculate from '../logic/calculate';

const useStyles = makeStyles({
  calWidth: {
    maxWidth: '375px',
  },
});

const Calculator = () => {
  const calculator = {
    total: null,
    next: null,
    operation: null,
  };

  const [calState, setCalState] = useState(calculator);
  const classes = useStyles();

  const handleClick = btnName => {
    const result = calculate(calState, btnName);
    setCalState(() => result);
  };

  const displayProperValue = (total, next) => {
    if (next) {
      return next;
    }
    return total;
  };
  return (
    <Grid container justify="center" direction="column" alignItems="center">
      <div className={classes.calWidth}>
        <Display result={displayProperValue(calState.total, calState.next)} />
        <ButtonPanel clickHandler={handleClick} />
      </div>
    </Grid>
  );
};

export default Calculator;
