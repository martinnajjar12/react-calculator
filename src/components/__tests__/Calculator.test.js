// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('Calculator page', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<Calculator />);
    expect(tree).toMatchSnapshot();
  });
});
