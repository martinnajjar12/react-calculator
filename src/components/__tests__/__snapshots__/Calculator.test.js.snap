// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page should match the snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="makeStyles-calWidth-1"
  >
    <div
      className="makeStyles-root-2"
    >
      0
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 makeStyles-topPanel-5 MuiButton-containedPrimary"
        disabled={false}
        name="AC"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          AC
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 makeStyles-topPanel-5 MuiButton-containedPrimary"
        disabled={false}
        name="+/-"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          +/-
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 makeStyles-topPanel-5 MuiButton-containedPrimary"
        disabled={false}
        name="%"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          %
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedSecondary"
        disabled={false}
        name="÷"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          ÷
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedPrimary"
        disabled={false}
        name="7"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          7
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedPrimary"
        disabled={false}
        name="8"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          8
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedPrimary"
        disabled={false}
        name="9"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          9
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedSecondary"
        disabled={false}
        name="×"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          ×
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedPrimary"
        disabled={false}
        name="4"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          4
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedPrimary"
        disabled={false}
        name="5"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          5
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedPrimary"
        disabled={false}
        name="6"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          6
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedSecondary"
        disabled={false}
        name="−"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          −
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedPrimary"
        disabled={false}
        name="1"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          1
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedPrimary"
        disabled={false}
        name="2"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          2
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedPrimary"
        disabled={false}
        name="3"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          3
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedSecondary"
        disabled={false}
        name="+"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          +
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 makeStyles-zeroButton-4 MuiButton-containedPrimary"
        disabled={false}
        name="0"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          0
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedPrimary"
        disabled={false}
        name="."
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          .
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-3 MuiButton-containedSecondary"
        disabled={false}
        name="="
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          =
        </span>
      </button>
    </div>
  </div>
</div>
`;
